### ISSUE TITLE
Tessellation possibly not working properly

### ISSUE THREAD

**Issue by MatthewKush** [turanszkij/WickedEngine#19]
Hi, great work on this, I'm always happy when I see this engine get updated. I'm learning a lot from you.

Anyway, whenever I try and enable tessellation in D3D11 and then proceed to use the tessellation multipler on a mesh, it turns black instead of being tessellated.

**Comment by turanszkij** [comment]
Hi, thanks for reporting this. I will take a look when I have some time. To be honest it was a very long time since I touched tessellation and I was kinda hoping no one would try anyway, but here we go. :)

**Comment by turanszkij** [comment]
I couldn't reproduce the exact problem you described, but there was an unbound constant buffer for a tessellator stage which I fixed. It should be working now.
To use tessellation, you can either tick the Tessellation enabled checkbox in the editor or call wiRenderer::SetTessellationEnabled(true) from application code. After that, meshes will be tessellated according to their tessellationfactor value.
Let me know if you are still having problems.

**Comment by MatthewKush** [comment]
lol...it's still happening yet you closed this... 

I'm sorry for not replying sooner. I'm using D3D11. It doesn't matter what I do, tessellation isn't working. I tend to reply on Sponza for testing. I added a displacement map for the lion head, and sure enough after allowing tessellation and selecting more than 1, the mesh turns black.

**Comment by turanszkij** [comment]
Sorry about that. What is your engine version number displayed at the top left corner?

**Comment by MatthewKush** [comment]
0.21.6

Sorry I was a jerk about it, won't happen again.

I'm not sure whats causing it, but basically: I make sure tessellation is checked in the renderer, I go to tessellate a mesh that I have given a displacement map to. Then I choose a tessellation factor other than zero for any mesh and it immediately turns the mesh and material black.

**Comment by turanszkij** [comment]
So I've game it an other look. First, displacement map was not working for tessellation (I removed it long ago). I readded it again, but has some todos: need to add parameter to control displacement strength, and need to add proper texture LODing which is not trivial to do inside domain shader. 
For the time being, tesellation supports contour tessellation better, which rounds mesh geometry edges.

I also could reproduce the flickering on one nvidia GPU, and unfortunately it looks like driver issue to me right now. It only happens when there are a certain amount of draw calls/instances for some reason which should not affect it. The debug layer is not giving me any warnings so I'm not sure how would I approach it. Also it works fine in Nsight graphics debugger. I will maybe contact Nvidia with a repro sample some time. 

I will keep this open for the time being. Thanks for reporting this again!
